metrotranscom/doorway
# fix: add hba jurisdiction data transfers

This PR addresses [#715](https://github.com/housingbayarea/bloom/issues/715)


## Description

Initial script for moving over data from SMC. This adds two scripts, one for copying over the base jurisdiction data such as AMI charts and preferences and the other is for copying over listings/units.

Notes:
- Images and lottery results will be manually ported over by product. That way the images will be moved to s3.
- Requested approver field is not going to get ported over since this doesn't port over the users at this point. Talked with product and that is ok with them.

## How Can This Be Tested/Reviewed?

This is how I've been testing it:
- Do a reseed locally
- Start up the app locally and go to the swagger api `http://localhost:3100/api`.
- Login via the api to the admin account `http://localhost:3100/api#/auth/login`.
- Go to Heroku and get the DB credentials for the HBA staging environment `https://data.heroku.com/datastores/fd968c98-f208-4066-b8cf-7d3a6bf8ff03#administration`
- Use the credentials from above to run the first script `http://localhost:3100/api#/scriptRunner/transferJurisdictionData`. I've been using "San Mateo" as the jurisdiction but any of the three jurisdictions would work
- Verify that AMI charts and multiselect questions are in the database. You can also verify that they show up in the partner site (This will only work for San Mateo)
- Run the script for `http://localhost:3100/api#/scriptRunner/transferJurisdictionListingsData` using the same db credentials and jurisdiction used in the previous script
- Verify that the new Listings look identical in partners between HBA staging vs local partners


## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
